C3real-kill3r/H-digest
# Enable profile creation, update and deletion #167315760

**What does this PR do?**
- [x] create user profile at time of user creation
- [x] allow update of user profile
- [x] allow for creation, updating and deletion of disability entries
- [x] create custom permission for `IsOwnerOrReadOnly`

**How to test**
- clone this repo `git clone https://github.com/C3real-kill3r/NCPWD.git `
- checkout to this branch `git checkout ft-enable-profile-creation-167315760`
- run the migration `python manage.py migrate`
- run the server `python manage.py runserver`
- create a user by sending a  post request to `http://127.0.0.1:8000/api/users/` with the fields
```
{
	"user":{
	"first_name":"john",
	"last_name":"Doe",
	"username":"username",
	"email":"email@email.com",
	"password":"ValidPassword"
}}
```
- send a get request to `http://127.0.0.1:8000/api/profiles/` to get the created profile
- update it by sending a put request to `http://127.0.0.1:8000/api/profiles/<id from above response>` with the following fields
```
{
  "id": <id>,
  "phone": "number",
  "location": "number",
  "national_id": "id_number",
  "date_of_birth": "DOB",
  "nationality": "lkenyan",
  "sex": "MALE/FEMALE",
  "user": user_id,
  "disabilities": []
}
```

Any pertinent information that should be considered

**What are the relevant pivotal tracker stories?**
[#167315760](https://www.pivotaltracker.com/story/show/167315760)

**Questions:**

If something is unclear or you want some questions to be addressed by your peers, mention them here
